test/tests/val-2a/val-2a_TMAP4.i

```diff
@@ -0,0 +1,81 @@
+nx_scale = 5
+
+[Mesh]
+  [cartesian_mesh_TMAP4]
+    type = CartesianMeshGenerator
+    dim = 1
+    #     num
+    dx = '${fparse 5 * ${units 4e-9 m -> mum}}  ${units 1e-8 m -> mum}  ${units 1e-7 m -> mum}
+          ${units 1e-6 m -> mum}                ${units 1e-5 m -> mum}  ${fparse 10 * ${units 4.88e-5 m -> mum}}'
+    ix = '${fparse 5 * ${nx_scale}}             ${nx_scale}             ${nx_scale}
+          ${nx_scale}                           ${nx_scale}             ${fparse 10 * ${nx_scale}}'
+  []
+[]
+
+[Functions]
+  ################# TMAP4
+  [Kd_left_func]
+    type = ParsedFunction
+    expression = '${dissociation_coefficient_parameter_enclos1} * (1 - 0.9999 * exp(-6e-5 * t))'
+  []
+
+  [Kr_left_func]
+    type = ParsedFunction
+    expression = '${recombination_coefficient_parameter_enclos1_TMAP4} * (1 - 0.9999 * exp(-6e-5 * t))'
+  []
+
+  [pressure_func]
+    type = ParsedFunction
+    expression = 'if(t<5820.0, ${pressure_high},
+                  if(t<9056.0, ${pressure_low},
+                  if(t<12062,  ${pressure_high},
+                  if(t<14572,  ${pressure_low},
+                  if(t<17678,  ${pressure_high}, ${pressure_low})))))'
+  []
+
+  [surface_flux_func]
+    type = ParsedFunction
+    expression = 'if(t<5820.0, ${flux_high},
+                  if(t<9056.0, ${flux_low},
+                  if(t<12062,  ${flux_high},
+                  if(t<14572,  ${flux_low},
+                  if(t<17678,  ${flux_high}, ${flux_low}))))) * 0.75'
+  []
+
+  [source_distribution]
+    type = ParsedFunction
+    symbol_names = 'width depth'
+    symbol_values = '${units 2.4e-9 m -> mum} ${units 14e-9 m -> mum}'
```

Comment:
These values could go to the top of the input files and you would not have to use `symbol_names` and `symbol_values`.